link.go

```diff
@@ -217,6 +217,9 @@ func attachLink(s *Session, r *Receiver, opts []LinkOption) (*link, error) {
 	}
 
 	if isReceiver {
+		if l.Source == nil {
```

Comment:
Ah, this would be the bug you mentioned in the PR message?

Reply:
Yeah, I improved on the fix in a later commit and applied the same to `l.Target`